Failed to reproduce "Generating a client in an Xcode project" example

Commit hash: 37255b7d12e7de407e8d38dd6456f4f144d44e6e

Context: I just wanted to try [example](https://swiftpackageindex.com/apple/swift-openapi-generator/0.1.1/tutorials/swift-openapi-generator/clientxcode), but build fails.

Steps to reproduce:

1. Create new Xcode project (iOS app, Swift, SwiftUI)
2. Follow steps from [tutorial](https://swiftpackageindex.com/apple/swift-openapi-generator/0.1.1/tutorials/swift-openapi-generator/clientxcode) including [Section 1](https://swiftpackageindex.com/apple/swift-openapi-generator/0.1.1/tutorials/swift-openapi-generator/clientxcode#Optional-Downloading-and-running-the-server-locally-for-testing) and [Section 2](https://swiftpackageindex.com/apple/swift-openapi-generator/0.1.1/tutorials/swift-openapi-generator/clientxcode#Optional-Creating-a-new-Xcode-target)
3. Build fails on Step 3 of [Section 3](https://swiftpackageindex.com/apple/swift-openapi-generator/0.1.1/tutorials/swift-openapi-generator/clientxcode#Configuring-your-target-to-use-the-Swift-OpenAPI-Generator-plugin)

XCode Version 14.3 (14E222b)

```
$ swift --version
swift-driver version: 1.75.2 Apple Swift version 5.8 (swiftlang-5.8.0.124.2 clang-1403.0.22.11.100)
Target: arm64-apple-macosx13.0
```

```
$uname -a
Darwin MacBook-Pro-Aleksej.local 22.5.0 Darwin Kernel Version 22.5.0: Mon Apr 24 20:52:24 PDT 2023; root:xnu-8796.121.2~5/RELEASE_ARM64_T6000 arm64
```

Hi @czechboy0, sure.

```Ld /Users/alexey/Library/Developer/Xcode/DerivedData/swift-openapi-ios-feadsqjpxalfivbzkcxwrdkhfssd/Build/Products/Debug-iphonesimulator/GeneratedClient.framework/GeneratedClient normal (in target 'GeneratedClient' from project 'swift-openapi-ios')
    cd /Users/alexey/Developer/ios/examples/swift-openapi/swift-openapi-ios
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -Xlinker -reproducible -target arm64-apple-ios16.4-simulator -dynamiclib -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator16.4.sdk -L/Users/alexey/Library/Developer/Xcode/DerivedData/swift-openapi-ios-feadsqjpxalfivbzkcxwrdkhfssd/Build/Intermediates.noindex/EagerLinkingTBDs/Debug-iphonesimulator -L/Users/alexey/Library/Developer/Xcode/DerivedData/swift-openapi-ios-feadsqjpxalfivbzkcxwrdkhfssd/Build/Products/Debug-iphonesimulator -F/Users/alexey/Library/Developer/Xcode/DerivedData/swift-openapi-ios-feadsqjpxalfivbzkcxwrdkhfssd/Build/Intermediates.noindex/EagerLinkingTBDs/Debug-iphonesimulator -F/Users/alexey/Library/Developer/Xcode/DerivedData/swift-openapi-ios-feadsqjpxalfivbzkcxwrdkhfssd/Build/Products/Debug-iphonesimulator -filelist /Users/alexey/Library/Developer/Xcode/DerivedData/swift-openapi-ios-feadsqjpxalfivbzkcxwrdkhfssd/Build/Intermediates.noindex/swift-openapi-ios.build/Debug-iphonesimulator/GeneratedClient.build/Objects-normal/arm64/GeneratedClient.LinkFileList -install_name @rpath/GeneratedClient.framework/GeneratedClient -Xlinker -rpath -Xlinker @executable_path/Frameworks -Xlinker -rpath -Xlinker @loader_path/Frameworks -dead_strip -Xlinker -object_path_lto -Xlinker /Users/alexey/Library/Developer/Xcode/DerivedData/swift-openapi-ios-feadsqjpxalfivbzkcxwrdkhfssd/Build/Intermediates.noindex/swift-openapi-ios.build/Debug-iphonesimulator/GeneratedClient.build/Objects-normal/arm64/GeneratedClient_lto.o -Xlinker -export_dynamic -Xlinker -no_deduplicate -Xlinker -objc_abi_version -Xlinker 2 -fobjc-link-runtime -L/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/iphonesimulator -L/usr/lib/swift -Xlinker -add_ast_path -Xlinker /Users/alexey/Library/Developer/Xcode/DerivedData/swift-openapi-ios-feadsqjpxalfivbzkcxwrdkhfssd/Build/Intermediates.noindex/swift-openapi-ios.build/Debug-iphonesimulator/GeneratedClient.build/Objects-normal/arm64/GeneratedClient.swiftmodule -Xlinker -no_adhoc_codesign -compatibility_version 1 -current_version 1 -Xlinker -dependency_info -Xlinker /Users/alexey/Library/Developer/Xcode/DerivedData/swift-openapi-ios-feadsqjpxalfivbzkcxwrdkhfssd/Build/Intermediates.noindex/swift-openapi-ios.build/Debug-iphonesimulator/GeneratedClient.build/Objects-normal/arm64/GeneratedClient_dependency_info.dat -o /Users/alexey/Library/Developer/Xcode/DerivedData/swift-openapi-ios-feadsqjpxalfivbzkcxwrdkhfssd/Build/Products/Debug-iphonesimulator/GeneratedClient.framework/GeneratedClient

Undefined symbols for architecture arm64:
  "_$s10Foundation3URLV14OpenAPIRuntimeE010validatingc9APIServerB0ACSS_tKcfC", referenced from:
      _$s15GeneratedClient7ServersO7server110Foundation3URLVyKFZ in Types.o
      _$s15GeneratedClient7ServersO7server210Foundation3URLVyKFZ in Types.o
  "_$s14OpenAPIRuntime10HTTPMethodV3getACvgZ", referenced from:
      _$s15GeneratedClient0B0V11getGreetingyAA10OperationsOADO6OutputOAG5InputVYaKF14OpenAPIRuntime7RequestVAKKXEfU_ in Client.o
  "_$s14OpenAPIRuntime11HeaderFieldVN", referenced from:
      _$s14OpenAPIRuntime7RequestV4path5query6method12headerFields4bodyACSS_SSSgAA10HTTPMethodVSayAA11HeaderFieldVG10Foundation4DataVSgtcfcfA2_ in Client.o
  "_$s14OpenAPIRuntime15UniversalClientV4send5input12forOperation10serializer12deserializerq_x_SSAA7RequestVxKXEq_AA8ResponseVKXEtYaKr0_lF", referenced from:
      _$s15GeneratedClient0B0V11getGreetingyAA10OperationsOADO6OutputOAG5InputVYaKFTY0_ in Client.o
  "_$s14OpenAPIRuntime15UniversalClientV4send5input12forOperation10serializer12deserializerq_x_SSAA7RequestVxKXEq_AA8ResponseVKXEtYaKr0_lFTu", referenced from:
      _$s15GeneratedClient0B0V11getGreetingyAA10OperationsOADO6OutputOAG5InputVYaKFTY0_ in Client.o
  "_$s14OpenAPIRuntime15UniversalClientV9serverURL13configuration9transport11middlewaresAC10Foundation0F0V_AA13ConfigurationVAA0D9Transport_pSayAA0D10Middleware_pGtcfC", referenced from:
      _$s15GeneratedClient0B0V9serverURL13configuration9transport11middlewaresAC10Foundation0D0V_14OpenAPIRuntime13ConfigurationVAK0B9Transport_pSayAK0B10Middleware_pGtcfC in Client.o
  "_$s14OpenAPIRuntime15UniversalClientVMa", referenced from:
      _$s15GeneratedClient0B0V9serverURL13configuration9transport11middlewaresAC10Foundation0D0V_14OpenAPIRuntime13ConfigurationVAK0B9Transport_pSayAK0B10Middleware_pGtcfC in Client.o
      _$s14OpenAPIRuntime15UniversalClientVWOb in Client.o
      _$s15GeneratedClient0B0VWOc in Client.o
      _$s15GeneratedClient0B0VWOh in Client.o
      _$s15GeneratedClient0B0V9converter33_627C69B1B4EBE1C19DF11ABF1978AC15LL14OpenAPIRuntime9ConverterVvg in Client.o
      _$s15GeneratedClient0B0V11getGreetingyAA10OperationsOADO6OutputOAG5InputVYaKF in Client.o
      _$s14OpenAPIRuntime15UniversalClientVWOc in Client.o
      ...
  "_$s14OpenAPIRuntime15UniversalClientVMn", referenced from:
      _symbolic _____ 14OpenAPIRuntime15UniversalClientV in Client.o
  "_$s14OpenAPIRuntime18_StringConvertibleMp", referenced from:
      _symbolic ______p 14OpenAPIRuntime18_StringConvertibleP in Client.o
  "_$s14OpenAPIRuntime19UndocumentedPayloadV23__derived_struct_equalsySbAC_ACtFZ", referenced from:
      _$s15GeneratedClient10OperationsO11getGreetingO6OutputO21__derived_enum_equalsySbAG_AGtFZ in Types.o
  "_$s14OpenAPIRuntime19UndocumentedPayloadVACycfC", referenced from:
      _$s15GeneratedClient0B0V11getGreetingyAA10OperationsOADO6OutputOAG5InputVYaKFAI14OpenAPIRuntime8ResponseVKXEfU0_ in Client.o
  "_$s14OpenAPIRuntime19UndocumentedPayloadVMn", referenced from:
      _symbolic Si10statusCode______t 14OpenAPIRuntime19UndocumentedPayloadV in Types.o
  "_$s14OpenAPIRuntime19UndocumentedPayloadVN", referenced from:
      _$s15GeneratedClient10OperationsO11getGreetingO6OutputO4hash4intoys6HasherVz_tF in Types.o
      _$s14OpenAPIRuntime19UndocumentedPayloadVACSHAAWl in Types.o
  "_$s14OpenAPIRuntime19UndocumentedPayloadVSHAAMc", referenced from:
      _$s14OpenAPIRuntime19UndocumentedPayloadVACSHAAWl in Types.o
  "_$s14OpenAPIRuntime25suppressMutabilityWarningyyxzlF", referenced from:
      _$s15GeneratedClient0B0V11getGreetingyAA10OperationsOADO6OutputOAG5InputVYaKF14OpenAPIRuntime7RequestVAKKXEfU_ in Client.o
  "_$s14OpenAPIRuntime7RequestV4path5query6method12headerFields4bodyACSS_SSSgAA10HTTPMethodVSayAA11HeaderFieldVG10Foundation4DataVSgtcfC", referenced from:
      _$s15GeneratedClient0B0V11getGreetingyAA10OperationsOADO6OutputOAG5InputVYaKF14OpenAPIRuntime7RequestVAKKXEfU_ in Client.o
  "_$s14OpenAPIRuntime7RequestVN", referenced from:
      _$s15GeneratedClient0B0V11getGreetingyAA10OperationsOADO6OutputOAG5InputVYaKF14OpenAPIRuntime7RequestVAKKXEfU_ in Client.o
  "_$s14OpenAPIRuntime9ConverterV18setQueryItemAsText2in4name5valueyAA7RequestVz_SSxSgtKAA18_StringConvertibleRzlF", referenced from:
      _$s15GeneratedClient0B0V11getGreetingyAA10OperationsOADO6OutputOAG5InputVYaKF14OpenAPIRuntime7RequestVAKKXEfU_ in Client.o
  "_$s14OpenAPIRuntime9ConverterV19renderedRequestPath8template10parametersS2S_SayAA18_StringConvertible_pGtKF", referenced from:
      _$s15GeneratedClient0B0V11getGreetingyAA10OperationsOADO6OutputOAG5InputVYaKF14OpenAPIRuntime7RequestVAKKXEfU_ in Client.o
  "_$s14OpenAPIRuntime9ConverterV20setHeaderFieldAsText2in4name5valueySayAA0eF0VGz_SSxSgtKAA18_StringConvertibleRzlF", referenced from:
      _$s15GeneratedClient0B0V11getGreetingyAA10OperationsOADO6OutputOAG5InputVYaKF14OpenAPIRuntime7RequestVAKKXEfU_ in Client.o
  "_$s14OpenAPIRuntime9ConverterV21getResponseBodyAsJSON_4from12transformingq_xm_10Foundation4DataVq_xXEtKSeRzr0_lF", referenced from:
      _$s15GeneratedClient0B0V11getGreetingyAA10OperationsOADO6OutputOAG5InputVYaKFAI14OpenAPIRuntime8ResponseVKXEfU0_ in Client.o
  "_$s14OpenAPIRuntime9ConverterV28validateContentTypeIfPresent2in9substringySayAA11HeaderFieldVG_SStKF", referenced from:
      _$s15GeneratedClient0B0V11getGreetingyAA10OperationsOADO6OutputOAG5InputVYaKFAI14OpenAPIRuntime8ResponseVKXEfU0_ in Client.o
  "_$sSS14OpenAPIRuntime18_StringConvertibleAAWP", referenced from:
      _$s15GeneratedClient0B0V11getGreetingyAA10OperationsOADO6OutputOAG5InputVYaKF14OpenAPIRuntime7RequestVAKKXEfU_ in Client.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

--- a/Sources/swift-openapi-generator/Documentation.docc/Tutorials/ClientXcode.tutorial
+++ b/Sources/swift-openapi-generator/Documentation.docc/Tutorials/ClientXcode.tutorial
@@ -101,7 +101,7 @@
             @Step {
                 Repeat the same steps two more times, with the packages `https://github.com/apple/swift-openapi-runtime` and `https://github.com/apple/swift-openapi-urlsession`.
                 
-                This time do check the library products to be added to the GeneratedClient target.
+                This time, do check the library products to be added to the **GeneratedClient target**. Note, this might not be the default target Xcode offers to add the libraries to.
             }
             @Step {
                 To finish configuring the build plugin in your target, navigate to the Build Phases tab of the GeneratedClient in the Project Editor, and expand the Run Build Tool Plug-ins section.

